hermeto/core/package_managers/gomod.py

```diff
@@ -688,6 +688,13 @@ def fetch_gomod_source(request: Request) -> RequestOutput:
     if not subpaths:
         return RequestOutput.empty()
 
+    if not (installed_toolchains := _list_installed_toolchains()):
+        raise FetchError(
+            "Could not find any installed Go toolchains in known locations",
+            solution="Please make sure at least one go toolchain is installed in the system",
+        )
+    go: Optional[Go] = next(iter(installed_toolchains))
```

Comment:
Two observations here, first, why `Optional`? _list_installed_toolchains() returns a set of Go objects which are not None by definition. Second, since we already know that `installed_toolchains` is not empty and we don't care about the order why not just `.pop()`? _Edit_: having written that I realized that this is temporary logic. Keeping the comment just in case